### DISCUSSION TITLE
Unable to compile diesel, fails at openssl

### DISCUSSION THREAD

**Discussion by FadyQJ** [diesel-rs/diesel#4291] [Q&A] [answered]
## Setup

I am using Windows 10, created a new project, was able to run migrations but debugging is crashing.

### Versions

- Rust: 1.81.0
- Diesel: 2.2.4 & 2.2.0 (tried both)
- Database: postgres
- Operating System: Windows 10

### Feature Flags

- postgres, serde_json, chrono

## Problem Description

I've spent 2 days so far on this, I hope it is not some stupid issue but my project is having issues working with Postgres.
I am trying to connect to an AWS database but haven't been able to test anything yet.

### Steps to reproduce

If I use diesel = { workspace = true, features = ["postgres", "serde_json", "chrono"] }
Code compiles but I get `(exit code: 0xc0000135, STATUS_DLL_NOT_FOUND)` while debugging which I have so far understood to be openssl not being loaded dynamically.

If I add
pq-sys = { version = "0.6", features = ["bundled"] }
openssl-sys = { version = "0.9.100", features = ["vendored"] } 

I get
```
running "perl" "./Configure" "--prefix=F:/Projects/q-lib/target/debug/build/openssl-sys-9ca0df653092ba46/out/openssl-build/install" "--openssldir=SYS$MANAGER:[OPENSSL]" "no-dso" "no-shared" "no-ssl3" "no-tests" "no-comp" "no-zlib" "no-zlib-dynamic" "--libdir=lib" "no-md2" "no-rc5" "no-weak-ssl-ciphers" "no-camellia" "no-idea" "no-seed" "no-capieng" "no-asm" "VC-WIN64A"

--- stderr
  Can't locate Locale/Maketext/Simple.pm in @INC (you may need to install the Locale::Maketext::Simple module) (@INC entries checked: /f/Projects/q-lib/target/debug/build/openssl-sys-9ca0df653092ba46/out/openssl-build/build/src/util/perl /usr/lib/perl5/site_perl /usr/share/perl5/site_perl /usr/lib/perl5/vendor_perl /usr/share/perl5/vendor_perl /usr/lib/perl5/core_perl /usr/share/perl5/core_perl /f/Projects/q-lib/target/debug/build/openssl-sys-9ca0df653092ba46/out/openssl-build/build/src/external/perl/Text-Template-1.56/lib) at /usr/share/perl5/core_perl/Params/Check.pm line 6.
  BEGIN failed--compilation aborted at /usr/share/perl5/core_perl/Params/Check.pm line 6.
  Compilation failed in require at /usr/share/perl5/core_perl/IPC/Cmd.pm line 59.
  BEGIN failed--compilation aborted at /usr/share/perl5/core_perl/IPC/Cmd.pm line 59.
  Compilation failed in require at /f/Projects/q-lib/target/debug/build/openssl-sys-9ca0df653092ba46/out/openssl-build/build/src/util/perl/OpenSSL/config.pm line 19.
  BEGIN failed--compilation aborted at /f/Projects/q-lib/target/debug/build/openssl-sys-9ca0df653092ba46/out/openssl-build/build/src/util/perl/OpenSSL/config.pm line 19.
  Compilation failed in require at ./Configure line 23.
  BEGIN failed--compilation aborted at ./Configure line 23.
  thread 'main' panicked at C:\Users\Kue\.cargo\registry\src\index.crates.io-6f17d22bba15001f\openssl-src-300.3.2+3.3.2\src/lib.rs:633:9:

  Error configuring OpenSSL build:
      Command: "perl" "./Configure" "--prefix=F:/Projects/q-lib/target/debug/build/openssl-sys-9ca0df653092ba46/out/openssl-build/install" "--openssldir=SYS$MANAGER:[OPENSSL]" "no-dso" "no-shared" "no-ssl3" "no-tests" "no-comp" "no-zlib" "no-zlib-dynamic" "--libdir=lib" "no-md2" "no-rc5" "no-weak-ssl-ciphers" "no-camellia" "no-idea" "no-seed" "no-capieng" "no-asm" "VC-WIN64A"
      Exit status: exit code: 2
```

## What I tried

- Installing / reinstalling Postgres client 
- Installing vcpkg and installing openssl:x64-windows-static
- Installing openSSSL
- Adding lib paths so diesel can find them dynamically

I understand it may be an issue outside the diesel package.

**Comment by weiznich** [comment, marked as answer]
I've converted your issue to an discussion entry as it's not a bug in diesel, but an issue with your local development environment. Please do fill future questions in the discussion part of the repository instead, as the issue part is strictly for bugs only.

That written: It's always hard to help people with such environment issues, as they only depend on your local environment and you are the only person that has access to that environment.

I can point to diesels CI config as we also test things on windows there: https://github.com/diesel-rs/diesel/blob/master/.github/workflows/ci.yml#L153-L162. That works in CI, so if you follow that setup step by step you should be able to get it up and running.

> If I add
> pq-sys = { version = "0.6", features = ["bundled"] }
> openssl-sys = { version = "0.9.100", features = ["vendored"] }
>
> I get

This build error looks like you miss some perl module to build openssl. See the error message for more details:

> Can't locate Locale/Maketext/Simple.pm in INC **(you may need to install the Locale::Maketext::Simple module)**

**Reply by FadyQJ** [reply]
Thank you so much!

I saw
`Can't locate Locale/Maketext/Simple.pm in INC (you may need to install the Locale::Maketext::Simple module)`

But thought I was doing something wrong because what does perl have to do with this :D
